mitmproxy/net/tcp.py

```diff
@@ -306,9 +308,15 @@ class Address(serializable.Serializable):
         ipv6 information.
     """
 
-    def __init__(self, address, use_ipv6=False):
+    def __init__(self, address, use_ipv6=None):
         self.address = tuple(address)
-        self.use_ipv6 = use_ipv6
+        if use_ipv6 is None:
+            try:
+                self.use_ipv6 = isinstance(ip_address(address[0]), IPv6Address)
```

Comment:
Also here's a place where I try to guess a protocol based on supplied address, so "-b ::1" binds to correct address without need to speficy `-6` option.